source/Server.Extensibility/HostServices/Model/WorkerPools/WorkerPoolId.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using Octopus.Server.Extensibility.TinyTypes;
+
+namespace Octopus.Server.Extensibility.HostServices.Model.WorkerPools
+{
+    public class WorkerPoolId : CaseInsensitiveStringTinyType
+    {
+        public WorkerPoolId(string value) : base(value)
+        {
+            if (string.IsNullOrWhiteSpace(value))
+                throw new ArgumentException(nameof(value),
+                    "Worker pool IDs or Names should never be null or empty. If the value you want to communicate is null/empty/whitespace then " +
+                    "just pass a null rather than a tiny type wrapping it.");
```

Comment:
Can we recommend they create a `NullWorkerPoolId` instance instead? (ie NullObject pattern)... 
I'd hate to see us take 2 steps forwards (with tinytypes) but then double down on NullRefExceptions

Reply:
I think that one's a beer o'clock discussion. Happy to have it, though :)